code/modules/reagents/chemistry/reagents/toxin_reagents.dm

```diff
@@ -214,6 +214,11 @@
 	ph = 12
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
 
+/datum/reagent/toxin/carpotoxin/on_mob_add(mob/living/L, amount)
```

Comment:
There's already a system for species-dependant reagent processing. I'd use that instead.

Reply:
For reference, it's the ```handle_chemicals``` proc in the species datum. Have a look at plasmamen if you need to figure out how that looks.

Reply:
Changed this to handle_chemicals, tested it, works. Thank you!